test/suite/test_timestamp22.py

```diff
@@ -0,0 +1,444 @@
+                elif self.do_illegal():
+                    this_commit_ts = self.stable_ts - 1
+                else:
+                    # It's possible this will succeed, we'll check below.
+                    this_commit_ts = self.gen_ts(commit_ts)
+                config += ',commit_timestamp=' + timestamp_str(this_commit_ts)
+
+            if this_commit_ts >= 0:
+                if this_commit_ts < running_commit_ts:
+                    ok = False
+                if this_commit_ts < self.stable_ts:
+                    ok = False
+                if this_commit_ts < self.oldest_ts:
+                    ok = False
+            if not ok:
+                this_commit_ts = -1
+            if this_commit_ts >= 0:
+                running_commit_ts = this_commit_ts
+            return (ok, config, running_commit_ts)
+
+        session = self.session
+        needs_rollback = False
+        prepare_config = None
+        commit_config = 'commit_timestamp=' + timestamp_str(commit_ts)
+        tstxn1_config = ''
+        tstxn2_config = ''
+
+        ok_commit = do_prepare or not self.do_illegal()
+        ok_prepare = True
+        ok_tstxn1 = True
+        ok_tstxn2 = True
+
+        # Occasionally put a durable timestamp on a commit without a prepare,
+        # that will be an error.
+        if do_prepare or not ok_commit:
+            commit_config += ',durable_timestamp=' + timestamp_str(durable_ts)
+        cursor = session.open_cursor(self.uri)
+        prepare_ts = self.gen_ts(commit_ts)
+        prepare_config = 'prepare_timestamp=' + timestamp_str(prepare_ts)
+        begin_config = '' if read_ts < 0 else 'read_timestamp=' + timestamp_str(read_ts)
+
+        # We might do a timestamp_transaction calls either before/after inserting
+        # values, or both.
+        do_tstxn1 = (self.rand.rand32() % 10 == 0)
+        do_tstxn2 = (self.rand.rand32() % 10 == 0)
+
+        # Keep track of the commit timestamp that we'll set through the transaction.
+        # If it decreases, it will trigger an error.  At the final commit_transaction
+        # operation, we'll use the commit_ts.
+        running_commit_ts = -1
+        first_commit_ts = -1
+
+        if do_tstxn1:
+            (ok_tstxn1, tstxn1_config, running_commit_ts) = \
+                timestamp_txn_config(commit_ts, running_commit_ts)
+            if first_commit_ts < 0:
+                first_commit_ts = running_commit_ts
+
+        if do_tstxn2:
+            (ok_tstxn2, tstxn2_config, running_commit_ts) = \
+                timestamp_txn_config(commit_ts, running_commit_ts)
+            if first_commit_ts < 0:
+                first_commit_ts = running_commit_ts
+
+        # ODDITY: If any setting of the timestamp fail, then an ASSERT will be hit in prepare.
+        # Avoid this, it will crash the test suite when diagnostic mode is enabled, and without
+        # diagnostic mode, the prepare appears to succeed!  (Comment out the if to see it in action).
```

Comment:
The `if` is commented out. Also there are multiple `if` statements. It isn't clear which one you're referring to or if it supposed to be commented out.